homeassistant/components/group/switch.py

```diff
@@ -0,0 +1,175 @@
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize Switch Group config entry."""
+    registry = er.async_get(hass)
+    entities = er.async_validate_entity_ids(
+        registry, config_entry.options[CONF_ENTITIES]
+    )
+    async_add_entities(
+        [
+            SwitchGroup(
+                config_entry.entry_id,
+                config_entry.title,
+                entities,
+                config_entry.options[CONF_ALL],
+            )
+        ]
+    )
+
+
+class SwitchGroup(GroupEntity, SwitchEntity):
+    """Representation of a switch group."""
+
+    _attr_available = False
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        unique_id: str | None,
+        name: str,
+        entity_ids: list[str],
+        mode: bool | None,
+    ) -> None:
+        """Initialize a switch group."""
+        self._entity_ids = entity_ids
+
+        self._attr_name = name
+        self._attr_extra_state_attributes = {ATTR_ENTITY_ID: entity_ids}
+        self._attr_unique_id = unique_id
+        self.mode = any
+        if mode:
+            self.mode = all
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+
+        @callback
+        def async_state_changed_listener(event: Event) -> None:
+            """Handle child updates."""
+            self.async_set_context(event.context)
+            self.async_defer_or_update_ha_state()
+
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, self._entity_ids, async_state_changed_listener
+            )
+        )
+
+        await super().async_added_to_hass()
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Forward the turn_on command to all switches in the group."""
+        data = {ATTR_ENTITY_ID: self._entity_ids}
+        _LOGGER.debug("Forwarded turn_on command: %s", data)
+
+        await self.hass.services.async_call(
+            DOMAIN,
+            SERVICE_TURN_ON,
+            data,
+            blocking=True,
+            context=self._context,
+        )
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Forward the turn_off command to all switches in the group."""
+        data = {ATTR_ENTITY_ID: self._entity_ids}
+        await self.hass.services.async_call(
+            DOMAIN,
+            SERVICE_TURN_OFF,
+            data,
+            blocking=True,
+            context=self._context,
+        )
+
+    @callback
+    def async_update_group_state(self) -> None:
+        """Query all members and determine the switch group state."""
+        all_states = [self.hass.states.get(x) for x in self._entity_ids]
+        states: list[State] = list(filter(None, all_states))
+
+        # filtered_states are members currently in the state machine
+        filtered_states: list[str] = [x.state for x in all_states if x is not None]
```

Comment:
How is this different than `states`?

Reply:
Ah it's the `state` string. Can we just base this on `states` and not filter it again ?

Reply:
Done